pxtcompiler/emitter/service.ts

```diff
@@ -231,8 +235,10 @@ namespace ts.pxtc {
                 snippet: service.getSnippet(decl, attributes)
             }
 
-            if (stmt.kind == SK.GetAccessor)
+            if ((stmt.kind === SK.GetAccessor && !getDeclarationOfKind(decl.symbol, SK.SetAccessor)) ||
```

Comment:
Aren't you getting double the setter with this change? I think the `addCombined("set", s)` will be called now once for the getter and once for the setter?

Reply:
(anyway should be easy to fix, just check if `qName` already exists before `ex.combinedProperties.push(s.qName)`)

Reply:
Don't understand the concern here re: addCombined() being called for setter and getter. The behavior should be the same as the old code prior to this PR. The difference is that I was seeing setter blocks getting generated when the property was readonly. This change fixes that by adding,

```typescript
(stmt.kind === SK.PropertyDeclaration && isReadonly(stmt as Declaration))) 
```

And the above change,

```typescript
if ((stmt.kind === SK.GetAccessor && !getDeclarationOfKind(decl.symbol, SK.SetAccessor)) || 
```

is intended to shore up code that I felt was wrong to begin with. It appeared that symbol info was created and marked as read-only when a statement was of _GetAccessor_ kind. That seemed weird to me. I made this particular change blind as I've been unable to get a case where the emitter is running over non-declared properties. I'm happy to revert, the only change really needed is the fix to check for read-only declared properties, i.e. the check for the _PropertyDeclaration_ kind.

Reply:
I think the change to remove isReadOnly from GetAccessors when there is also a setter is a good one. However, previously `addCombined("get", ...)` was called for the getter, and `addCombined("set", ...)` (and `"change"`) was called for the setter. Now, both `addCombined("get", ...)` and `addCombined("set", ...)` will be called for the  getter, leading to duplicate entries in the list (as the behavior for setter will be unchanged). Please just check for duplicates before adding to `ex.combinedProperties`.

Reply:
I'll take a look.

Reply:
Ah, I understand now. You know what I should revert the GetAccessor change.
In looking at the code it assumes a qualified named symbol per getter and per setter in the SymbolInfo. I had not realized this. I assumed it combined the two into one SymbolInfo. Personally I think that'd be better, but that is more code change than just checking for dupes on adding to combinedProperties. In the interest of not churning the code I'll revert part of this.

Reply:
Thanks for clarifying your original review comment.

Reply:
Submitted an updated PR. I feel that if setting the GetAccessor symbol kind to read-only is a good one, and it fixes another issue, than that should be applied as an independent change in the interest of less churn on this PR.